blueprints/controllers/ikea_e1812/ikea_e1812.yaml

```
@@ -59,197 +70,279 @@ blueprint:
+          mode: slider
+          step: 10.0
 variables:
-  # convert input tags to variables, to be used in templates
-  integration: !input integration
-  button_long_loop: !input button_long_loop
-  button_double_press: !input button_double_press
-  helper_last_controller_event: !input helper_last_controller_event
-  # integration id used to select items in the action mapping
+  integration: !input 'integration'
+  button_long_loop: !input 'button_long_loop'
+  button_long_max_loops: !input 'button_long_max_loops'
+  button_double_press: !input 'button_double_press'
+  button_double_initial_press: !input 'button_double_initial_press'
+  action_button_long: !input 'action_button_long'
+  helper_last_controller_event: !input 'helper_last_controller_event'
+  double_press_delay: !input 'double_press_delay'
+  debounce_delay: !input 'debounce_delay'
+  # Subtracted debounce delay so that the timeout is really as set in double press delay but never go below minimum delay.
+  adjusted_double_press_delay: '{% if double_press_delay - debounce_delay < 100 %}100{% else %}
+                                {{ (double_press_delay - debounce_delay) | int }} {% endif %}'
   integration_id: '{{ integration | lower }}'
-  # mapping between actions and integrations
   actions_mapping:
     deconz:
-      button_short: [1002]
-      button_long: [1001]
-      button_release: [1003]
+      button_short:
+      - 1002
+      button_long:
+      - 1001
+      button_release:
+      - 1003
     zha:
-      button_short: ['on']
-      button_long: [move_with_on_off]
-      button_release: [stop]
+      button_short:
+      - 'on'
+      button_long:
+      - move_with_on_off
+      button_release:
+      - stop
     zigbee2mqtt:
-      button_short: ['on']
-      button_long: [brightness_move_up]
-      button_release: [brightness_stop]
-  # pre-choose actions for buttons based on configured integration
-  # no need to perform this task at automation runtime
+      button_short:
+      - 'on'
+      button_long:
+      - brightness_move_up
+      button_release:
+      - brightness_stop
   button_short: '{{ actions_mapping[integration_id]["button_short"] }}'
   button_long: '{{ actions_mapping[integration_id]["button_long"] }}'
   button_release: '{{ actions_mapping[integration_id]["button_release"] }}'
-  # build data to send within a controller event
-  controller_entity: !input controller_entity
-  controller_device: !input controller_device
-  controller_id: '{% if integration_id=="zigbee2mqtt" %}{{controller_entity}}{% else %}{{controller_device}}{% endif %}'
-# mode restart ensures hold actions are interrupted when the corresponding release action is received
+  integrations_with_debounce:
+  - deconz
+  - zha
+  controller_entity: !input 'controller_entity'
+  controller_device: !input 'controller_device'
+  controller_id: '{% if integration_id=="zigbee2mqtt" %}{{controller_entity}}{% else
+    %}{{controller_device}}{% endif %}'
```

```suggestion
  integrations_with_debounce: [deconz, zha]
  # build data to send within a controller event
  controller_entity: !input 'controller_entity'
  controller_device: !input 'controller_device'
  controller_id: '{% if integration_id=="zigbee2mqtt" %}{{controller_entity}}{% else %}{{controller_device}}{% endif %}'
  # mode restart ensures hold actions are interrupted when the corresponding release action is received
```

use inline array format, bring back comments